content/docs/proxy/google_serp_proxy/examples.md

```diff
@@ -292,27 +271,6 @@ console.log(data);
 </marked-tab>
 
 
-<marked-tab header="Node.js (got)" lang="javascript">
-import got from 'got';
-import { HttpsProxyAgent } from 'hpagent';
-
-const agent = {
-    https: new HttpsProxyAgent({
```

Comment:
Have you noticed that there's `https` but you're querying `http` instead? :D

Reply:
yup, I saw the docs of the agent, and I tried all options, it would either throw or return local (CZ) results

Reply:
Look what I found :joy:

![image](https://user-images.githubusercontent.com/36894700/184607045-c0710a6e-24d1-40ae-b301-46d0a7f9a926.png)

Reply:
I think I even tried the direct URL 🤔 will check it once again once I am back to my laptop 👌

Reply:
So - I had `q` in my latest tries, but it does not matter - both should work I think. The problem is that proxy is ignored. It supposed to return US results, but returning Czech ones. It's with `https`, with `http` I have this:
<img width="587" alt="Screenshot 2022-08-15 at 12 01 08" src="https://user-images.githubusercontent.com/18262743/184616155-3a89414c-0b3f-49e1-87ba-76e33c15fb01.png">

Reply:
Can you try the code I posted?

Reply:
welp, looks like it finally worked, wtf 🤯

Reply:
however - it still does not work with the `hpagent` from `got` docs. Fails with `code: 'ERR_GOT_REQUEST_ERROR', literally the only thing changed is the `agent`.

Reply:
It doesn't work with `hpagent` because the proxy doesn't support `CONNECT` method: https://github.com/delvedor/hpagent/blob/cc90c2b60a0b946574993d5598d93229ce452f72/index.js#L18

```
RequestError: Bad response: 400
Google query failed: The proxy group "GOOGLESERP" cannot be used over HTTPS connection.
```

The protocol is `http:` but the superproxy mistakenly thinks it's HTTPS.

---

Basically there are two ways to proxy a request.

1. We can make a GET request like normal but include the full absolute URI:

```
GET http://example.com/ HTTP/1.1
Host: example.com
...
```

and receive a response from the proxy server.

2. We can use the `CONNECT` method. It's meant for setting up TCP over HTTP.

```
CONNECT example.com:80 HTTP/1.1
Host: example.com:80
...

[raw TCP data]
```

It's used to connect to other servers, they don't have to be HTTP.

---

I'd solve this in a superproxy this way. If there's a CONNECT method, set up a dummy server (doesn't have to listen on any port), and redirect request/responses through that one.

Reply:
Not sure if this is resolved, but I think it would make sense not to have to get into this level of detail in the examples 😅

Reply:
got examples are removed, I was just trying to understand what am I doing wrong here :) I don't plan to push it further. But if I would need it in real life - I would just use another package for agent, or even just use axios instead of got all along 🤷‍♂️

Reply:
@mnmkng Since those agents are problematic, I think we could move the `proxyUrl` option to Got? Agents indeed may be too low-level for newbies.

Reply:
For now you could recommend just using `got-scraping`.

Reply:
Or we could implement CONNECT for SERP as [mentioned here](https://github.com/apify/apify-docs/pull/425#discussion_r946538384) - this would be much more user friendly.

Reply:
I think we would not put `got` there, old docs had `requestAsBrowser`, which is now replaced with `got-scraping`, so we are kinda emphasizing it already, the rest is just illustrative - to basically show that you could use our proxies with other libraries too.

As for `proxyUrl` for `got` - personally, I think it's a good idea, because imho it's much user-friendly :)